src/demo-app/stepper/stepper-demo.html

```diff
@@ -1,3 +1,50 @@
+<h2>Linear Vertical Stepper Demo</h2>
+<form [formGroup]="formGroup">
+  <md-vertical-stepper formArrayName="formArray">
+    <md-step formGroupName="0">
+      <ng-template mdStepLabel>Fill out your name</ng-template>
+      <md-input-container>
+        <input mdInput placeholder="First Name" formControlName="firstNameFormCtrl" required>
+        <md-error>This field is required</md-error>
+      </md-input-container>
+
+      <md-input-container>
+        <input mdInput placeholder="Last Name" formControlName="lastNameFormCtrl" required>
+        <md-error>This field is required</md-error>
+      </md-input-container>
+      <div>
+        <button md-button mdStepperNext type="button">Next</button>
+      </div>
+    </md-step>
+
+    <md-step formGroupName="1"
+             [disabled]="formArray.hasError('invalid step') ?
+             formArray.getError('invalid step').index <= 1 : false">
+      <ng-template mdStepLabel>
+        <div>Fill out your phone number</div>
+      </ng-template>
+      <md-input-container>
+        <input mdInput placeholder="Phone number" formControlName="phoneFormCtrl">
+        <md-error>This field is required</md-error>
+      </md-input-container>
+      <div>
+        <button md-button mdStepperPrevious type="button">Back</button>
+        <button md-button mdStepperNext type="button">Next</button>
+      </div>
+    </md-step>
+
+    <md-step
+        [disabled]="formArray.hasError('invalid step') ?
```

Comment:
@joejordanbrown The changed code no longer only checks for the previous step's validity. Now it uses the custom step validation function `stepValidator` to check for the validity of all the previous steps. If there's a step that is invalid, it will return the smallest index of the step that is invalid to compare with the index of the step that is being triggered.

@mmalerba : Waiting for @kara 's response on additional suggestions regarding this.